omniscidb/QueryBuilder/QueryBuilder.cpp

```diff
@@ -591,6 +591,28 @@ BuilderExpr BuilderExpr::singleValue() const {
   return {builder_, agg, name, true};
 }
 
+BuilderExpr BuilderExpr::topK(int count) const {
+  if (!expr_->type()->isNumber() && !expr_->type()->isDateTime()) {
+    throw InvalidQueryError() << "Unsupported type for topK aggregate: "
+                              << expr_->type()->toString();
+  }
+  if (count == 0) {
+    throw InvalidQueryError()
+        << "Non-zero integer argument is expected for topK aggregate. Provided: "
```

Comment:
I was choosing between introducing two aggregates `topK` and `bottomK` and using a single aggregate for both cases. I didn't want to have `== hdk::ir::AggType::kTopK || ==hdk::ir::AggType::kBottomK` everywhere instead of `== hdk::ir::AggType::kTopK`. Also, non-negative values would make me add additional parameters to runtime functions and `TargetInfo` for the type of aggregate. And I guess negative values would be expected to be accepted in the top-level Python API anyway. So I chose to have a single aggregate allowing negative parameters with `topK` and `bottomK` APIs on the builder level.